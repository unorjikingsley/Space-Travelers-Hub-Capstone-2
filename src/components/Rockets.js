import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRockets } from '../redux/rocket/rocketSlice';
import RocketItem from './RocketItem';
import { selectRockets } from '../redux/store';

const Rockets = () => {
  const dispatch = useDispatch();

  const { rockets, isLoading, error } = useSelector(selectRockets);

  useEffect(() => {
    if (rockets.length > 0) return;
    dispatch(fetchRockets());
  }, [dispatch, rockets.length]);

  if (error) {
    return (
      <div>
        Something went wrong:
        <br />
        &quot;
        {error}
        &quot;
      </div>
    );
  }

  return (
    <section>
      {isLoading ? (
        <div>loading...</div>
      ) : (
        <>
          <ul>
            {rockets.map((rocket) => (
              <RocketItem key={rocket.id} rocket={rocket} />
            ))}
          </ul>
        </>
      )}
    </section>
  );

  // const RocketItems = useSelector((state) => state.rockets);
  //   return (
  //     <ul>
  //       {RocketItems.map((item) => (
  //         <RocketItem
  //           id={item.id}
  //           key={item.rocket_id}
  //           name={item.rocket_name}
  //           description={item.description}
  //           images={item.flickr_images}
  //         />
  //       ))}

//     </ul>
//   );
};

export default Rockets;
