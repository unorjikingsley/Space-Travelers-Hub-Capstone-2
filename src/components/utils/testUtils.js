import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
// import Table from 'react-bootstrap/Table';
import store from '../../redux/store';

const withRouter = (component) => (
  <BrowserRouter>
    {component}
  </BrowserRouter>
);

const withProvider = (component, mockStore = store) => (
  <Provider store={mockStore}>
    {component}
  </Provider>
);

const withTable = (component) => (
  <table>
    <tbody>
      {component}
    </tbody>
  </table>
);

export { withProvider, withRouter, withTable };
