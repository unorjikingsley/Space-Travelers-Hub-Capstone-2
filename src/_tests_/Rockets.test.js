import React from 'react';
import renderer from 'react-test-renderer';
import Rockets from '../components/Rockets';
import { withProvider } from '../components/utils/testUtils';

test('renders RocketsPage correctly', () => {
  const rockets = [
    {
      id: 1,
      rocket_name: 'Falcon 9',
      description: 'A two-stage rocket designed and manufactured by SpaceX.',
      image: 'falcon9.jpg',
      reserved: false,
    },
  ];

  const tree = renderer
    .create(withProvider(<Rockets rockets={rockets} />))
    .toJSON();

  expect(tree).toMatchSnapshot();
});
