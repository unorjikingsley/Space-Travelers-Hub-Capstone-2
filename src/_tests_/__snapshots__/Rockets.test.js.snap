// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders RocketsPage correctly 1`] = `
<section>
  <div>
    loading...
  </div>
</section>
`;
